services/tables/src/main/java/com/linkedin/openhouse/tables/repository/impl/OpenHouseInternalRepositoryImpl.java

```diff
@@ -160,6 +166,19 @@ public TableDto save(TableDto tableDto) {
         table, fileIOManager, partitionSpecMapper, policiesMapper, tableTypeMapper);
   }
 
+  private boolean skipEligibilityCheck(
+      Map<String, String> existingTableProps, Map<String, String> newTableprops) {
+    TableType existingTableType =
+        TableType.valueOf(existingTableProps.get(getCanonicalFieldName(TABLE_TYPE_KEY)));
+    TableType newTableType =
+        TableType.valueOf(newTableprops.get(getCanonicalFieldName(TABLE_TYPE_KEY)));
+    return existingTableType == TableType.REPLICA_TABLE
+        && newTableType == TableType.PRIMARY_TABLE
+        && !existingTableProps
+            .get(getCanonicalFieldName(CLUSTER_ID))
+            .equals(newTableprops.get(getCanonicalFieldName(CLUSTER_ID)));
```

Comment:
can we just have this skip-logic as part of eligibility check? something like when the base table is replica, just skip the eligibility check?

Reply:
and, why is "cluster id" check required?

Reply:
cluster_id and primary_table type check is to confirm the request is coming from other cluster and from a Primary Table.
Writes are blocked for replica_table at Auth layer, so for OSS this check would ensure proper exceptions are surfaced for requests coming from same cluster with discrepancy in tableType property.

Reply:
I think it is okay to include clusterID in this check. One nit: Can we just move this check to updateEligbilitycheck method itself instead of having a new one?

Reply:
Would prefer to keep the conditions in a separate method considering more cases can come up in future. 
But yeah, can move the `if` clause inside the `updateEligbilitycheck`.